serve/serve.py

```diff
@@ -45,44 +54,58 @@ def __call__(self, request, response):
         return self.handler(request, response)
 
     def handle_request(self, request, response):
-        worker_path = replace_end(request.url_parts.path, ".worker.html", ".worker.js")
+        path = self._get_path(request.url_parts.path, True)
         meta = "\n".join(self._get_meta(request))
-        return """<!doctype html>
+        return self.wrapper % {"meta": meta, "path": path}
+
+    def _get_path(self, path, resource_path):
+        for item in self.path_replace:
+            if len(item) == 2:
+                src, dest = item
+            else:
+                assert len(item) == 3
+                src = item[0]
+                dest = item[2 if resource_path else 1]
+            if path.endswith(src):
+                path = replace_end(path, src, dest)
+        return path
+
+    def _get_meta(self, request):
+        path = self._get_path(filesystem_path(self.base_path, request, self.url_base), False)
+        with open(path, "rb") as f:
+            for key, value in read_script_metadata(f, js_meta_re):
+                replacement = self._meta_replacement(key, value)
+                if replacement:
+                    yield replacement
+
+    def _meta_replacement(self, key, value):
```

Comment:
I think I would prefer a `HtmlWrapperHandler` class spliced into the inheritance tree, making `WrapperHandler._get_meta()` abstract.